### PR TITLE
Besu Operator

### PR DESCRIPTION
**Pull request by Sumaid**
# Short Description

Kubernetes operator to provide an easier way of packaging, deploying, and managing Hyperledger Besu network.

Part of Hyperledger mentorship program : [Wiki Page](https://wiki.hyperledger.org/pages/viewpage.action?pageId=31195317)

Signed-off-by: Sumaid Syed <sumaidsyed@gmail.com>

### CODE REVIEW

**Comment by trbs** [comment]
+1 proposal lgtm

Can you ask Mark Wagner to confirm sponsorship here ?

**Comment by n1zyz** [comment]
I approve this as a sponsor.

File: besu-operator.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+3. Not scalable, no easy way of spreading containers among VM’s
+
+The solution to all these is the Kubernetes, which undoubtedly is the future of container orchestration. Kubernetes (also known as k8s or “kube”) is an open source container orchestration platform that automates many of the manual processes involved in deploying, managing, and scaling containerized applications. It has a large, rapidly growing ecosystem. Kubernetes services, support, and tools are widely available.
+
+Kubernetes is a good fit for blockchain networks often running on multiple clouds and on-premise. Blockchain networks require different types of nodes - validators, bootnodes, and normal network nodes, so these arrangements can be easily configured for a particular cluster.
+
+Currently it's possible to deploy Besu on Kubernetes using Helm Charts or in kubectl. The purpose of the project is to wrap everything under Kubernetes operator, so one can provide the operator the desired state of Besu, and let it reach & maintain that state.
+
+This project directly affects the Hyperledger Besu Project. The basic operator structure & framework could be used to develop operators for other Hyperledger products.
```
Comment (tkuhrt) [inline besu-operator.md:22]: Change `This project` to `This lab`

File: besu-operator.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   - If a container dies, nothing will restart it
+   - If a virtual machine crashes, no failure recovery
+2. No monitoring, alerts, etc.
+3. Not scalable, no easy way of spreading containers among VM’s
+
+The solution to all these is the Kubernetes, which undoubtedly is the future of container orchestration. Kubernetes (also known as k8s or “kube”) is an open source container orchestration platform that automates many of the manual processes involved in deploying, managing, and scaling containerized applications. It has a large, rapidly growing ecosystem. Kubernetes services, support, and tools are widely available.
+
+Kubernetes is a good fit for blockchain networks often running on multiple clouds and on-premise. Blockchain networks require different types of nodes - validators, bootnodes, and normal network nodes, so these arrangements can be easily configured for a particular cluster.
+
+Currently it's possible to deploy Besu on Kubernetes using Helm Charts or in kubectl. The purpose of the project is to wrap everything under Kubernetes operator, so one can provide the operator the desired state of Besu, and let it reach & maintain that state.
```
Comment (tkuhrt) [inline besu-operator.md:20]: Change `the project` to `the lab`

File: besu-operator.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+This project directly affects the Hyperledger Besu Project. The basic operator structure & framework could be used to develop operators for other Hyperledger products.
+
+Details : [Project Google Doc](https://docs.google.com/document/d/1GdBv62g8dsASe4QSESar5JZNzMq2gq5L5onEqJa2HFg/edit?usp=sharing)
```
Comment (tkuhrt) [inline besu-operator.md:24]: Change `Project` to `Lab`

File: besu-operator.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# Scope of Lab
+Deploying applications using plain Docker containers has many issues such as :
+
+1. No management layer on top of it
+   - If a container dies, nothing will restart it
+   - If a virtual machine crashes, no failure recovery
+2. No monitoring, alerts, etc.
+3. Not scalable, no easy way of spreading containers among VM’s
+
+The solution to all these is the Kubernetes, which undoubtedly is the future of container orchestration. Kubernetes (also known as k8s or “kube”) is an open source container orchestration platform that automates many of the manual processes involved in deploying, managing, and scaling containerized applications. It has a large, rapidly growing ecosystem. Kubernetes services, support, and tools are widely available.
```
Comment (tkuhrt) [inline besu-operator.md:16]: Change `the Kubernetes` to `Kubernetes`

File: besu-operator.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+Details : [Project Google Doc](https://docs.google.com/document/d/1GdBv62g8dsASe4QSESar5JZNzMq2gq5L5onEqJa2HFg/edit?usp=sharing)
+
+# Initial Committers
+- https://github.com/Sumaid
+
+# Sponsor
+- Mark Wagner (mwagner@redhat.com) - a TSC member
+
+# Pre-existing repository
```
Comment (tkuhrt) [inline besu-operator.md:32]: Will the intent be to create a lab repo from scratch and transfer the code manually from these repositories?

File: besu-operator.md (hunk 4 above)
Comment (lehors) [reply besu-operator.md:16]: One could argue that Kubernetes is not "the future of container orchestration". It already is the present! ;-)
Joke apart, I think it's kind of subjective statement isn't necessary. I would rephrase: "A solution to all these is Kubernetes." and proceed with the next sentence "Kubernetes (also known..."

File: besu-operator.md (hunk 5 above)
Comment (Sumaid) [reply besu-operator.md:32]: Yes

File: besu-operator.md (hunk 4 above)
Comment (Sumaid) [reply besu-operator.md]: @tkuhrt Thanks for the suggestions, I have made the changes.

**Comment by tkuhrt** [comment]
Moved besu-operator.md to labs folder in #140 
Created Repo: https://github.com/hyperledger-labs/besu-operator
Sent invite to @Sumaid and @n1zyz
